init_suite.py

```diff
@@ -11,6 +11,29 @@
 log = Log(__name__)
 
 
+def merge_dicts(dict1, dict2):
+    """
+    Returns dict1 by recursively merging dict2 into dict1
+
+    Args:
+        dict1 (dict):     The dictionary corressponding to test in <test_suite>.
+        dict2 (dict):     The dictionary corressponding to test in <overrides>.
+
+    Returns:
+        Dict -> dictionary after merging overrides dict into test_suite dict
+    """
+    if not isinstance(dict1, dict) or not isinstance(dict2, dict):
+        return dict2
+    for k in dict2:
+        if k in dict1:
+            dict1[k] = merge_dicts(dict1[k], dict2[k])
+        else:
+            dict1[k] = dict2[k]
+        if dict1[k] is None:
+            del dict1[k]
```

Comment:
IMO, its better to keep null valued properties in test suite config as we will anyways handle those null exceptions inside code.